src/core/src/energy/functions.rs

```diff
@@ -364,6 +364,50 @@ impl PairPotential for BornMayerHuggins {
     }
 }
 
+/// Morse potential.
+///
+/// The following potential expression is used: `V(r) = De * (1 - exp(-a (r - re))^2 ; where 'r' is the distance between the atoms, 
+/// 're' is the equilibrium bond distance, 'De' is the well depth, and 'a' controls the 'width' of the potential
+///
+/// # Examples
+///
+/// ```
+/// use lumol::energy::Potential;
+/// use lumol::energy::MorsePotential;
+///
+/// let potential = MorsePotential{a: 2.0, r0: 3.0, De: 4.0};
+/// assert_eq!(MorsePotential.energy(1.0), 3.611618461763754);
+/// assert_eq!(MorsePotential.force(2.5), 6.25);
+/// ```
+#[derive(Clone, Copy)]
+pub struct MorsePotential {
+    /// Width value
+    pub width: f64, 
+    /// Equilibrium bond distance
+    pub r0: f64,
+    /// Well depth value
+    pub depth: f64,
+}
+
+impl Potential for MorsePotential {  
+    fn energy(&self, r: f64) -> f64 { 
+        let rc = 1.0 - f64::exp(-self.width * (r - self.r0));
+        let rc2 = rc * rc;
+        self.depth * (rc2 - 1.0)
+    }
+
+    fn force(&self, r: f64) -> f64 { 
+        self.depth * (f64::exp(-2.0 * self.width * (r - self.r0)) - 2.0 * f64::exp(-self.width * (r - self.r0)))
```

Comment:
The force should be the derivative of the energy, and it does not look like it here. It calculated it and I got

```
2 * De * width * (1 - exp(-width * (r - re)) * exp(-width * (r - re))
```